protocol/x/vault/keeper/shares.go

```diff
@@ -0,0 +1,34 @@
+package keeper
+
+import (
+	"cosmossdk.io/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dydxprotocol/v4-chain/protocol/x/vault/types"
+)
+
+// GetTotalShares gets TotalShares for a vault.
+func (k Keeper) GetTotalShares(
+	ctx sdk.Context,
+	vaultId types.VaultId,
+) (val types.NumShares, exists bool) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), []byte(types.TotalSharesKeyPrefix))
+
+	b := store.Get(vaultId.ToStateKey())
+	if b == nil {
+		return val, false
+	}
+
+	k.cdc.MustUnmarshal(b, &val)
+	return val, true
+}
+
+// SetTotalShares sets TotalShares for a vault.
+func (k Keeper) SetTotalShares(
```

Comment:
I wonder if instead of having `SetTotalShares`, we should have `incrementTotalShares`, `decrementTotalShares`, or `resetTotalShares`(setting total shares to 0). It seems like we will never set totalShares to an arbitrary number, so `SetTotalShares` would require additional arithmetic.

nit: Should we panic on `totalShares` being a negative number?

Reply:
Yeah we can build these increment decrement on top of `Get` and `Set` for sure.

As for verifying whether `totalShares` is non-negative, perhaps we can add that later when building out `MsgDepositToVault`?

Reply:
chatted with Vincent offline about this check on negative. will change this to return error if negative